microceph/ceph/rgw.go

```diff
@@ -49,14 +49,45 @@ func EnableRGW(s common.StateInterface, port int) error {
 	return nil
 }
 
-// rgwUpdateDatabase records changes to the database.
-func rgwUpdateDatabase(s common.StateInterface) error {
+// DisableRGW disables the RGW service on the cluster.
+func DisableRGW(s common.StateInterface) error {
+	confPath := filepath.Join(os.Getenv("SNAP_DATA"), "conf")
+	dataPath := filepath.Join(os.Getenv("SNAP_COMMON"), "data")
+
+	if err := stopRGW(); err != nil {
```

Comment:
Another code style thing here which you may or may not want to continue, up to you.

In LXD's codebase, we don't allow the define+check if syntax, instead requiring the function call and assignment be done on the line prior to the if statement. The reason for that is mostly to avoid accidental, issue with variable scoping and to make it a bit easier when cherry-picking fixes and logic changes.

A lot of Go projects do use the more compact syntax just fine though, so it's definitely up to you, if you want to keep using it, let me know and I'll stop flagging it in reviews :)